entities/order.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative '../validation/errors'
+
+# class Order
+class Order
+  include Errors
+
+  attr_reader :book, :reader, :date
+
+  def initialize(book:, reader:, date:)
+    raise WrongClassError if !(book.is_a? Book) || !(reader.is_a? Reader)
```

Comment:
move all validations to (for example) `validate` method

Reply:
corrected